app/src/main/java/com/example/development/sakaiclientandroid/fragments/AllCoursesFragment.java

```diff
@@ -2,35 +2,60 @@
 
 import android.os.Bundle;
 import android.support.annotation.Nullable;
+import android.support.v4.app.FragmentActivity;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v4.widget.SwipeRefreshLayout;
+import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ExpandableListView;
 
 import com.example.development.sakaiclientandroid.NavActivity;
 import com.example.development.sakaiclientandroid.R;
+import com.example.development.sakaiclientandroid.api_models.gradebook.GradebookObject;
 import com.example.development.sakaiclientandroid.models.Course;
+import com.example.development.sakaiclientandroid.models.Term;
 import com.example.development.sakaiclientandroid.utils.DataHandler;
 import com.example.development.sakaiclientandroid.utils.custom.HomeFragmentExpandableListAdapter;
+import com.example.development.sakaiclientandroid.utils.custom.TreeViewItemClickListener;
+import com.example.development.sakaiclientandroid.utils.holders.CourseHeaderViewHolder;
+import com.example.development.sakaiclientandroid.utils.holders.GradeNodeViewHolder;
+import com.example.development.sakaiclientandroid.utils.holders.TermHeaderViewHolder;
 import com.example.development.sakaiclientandroid.utils.requests.RequestCallback;
+import com.example.development.sakaiclientandroid.utils.requests.SharedPrefsUtil;
+import com.unnamed.b.atv.model.TreeNode;
+import com.unnamed.b.atv.view.AndroidTreeView;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 
-public class AllCoursesFragment extends BaseFragment{
+import static com.example.development.sakaiclientandroid.NavActivity.COURSES_TAG;
 
+public class AllCoursesFragment extends BaseFragment{
 
-    private List<String> termHeaders;
-    private HashMap<String, List<Course>> headerToCourses;
 
+    private AndroidTreeView treeView;
     private SwipeRefreshLayout swipeRefreshLayout;
+    ArrayList<ArrayList<Course>> courses;
 
 
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        Bundle bun = getArguments();
+        try
+        {
+            ArrayList<ArrayList<Course>> courses = (ArrayList<ArrayList<Course>>) bun.getSerializable(COURSES_TAG);
+            this.courses = courses;
+            createTreeView(courses);
```

Comment:
Since `createTreeView` is called in `onCreateView` as well, isn't this unnecessary here? You can probably just omit this line since the action is performed in `onCreateView` anyways.

Reply:
oh yea good catch